src/test/java/tests/DetailMessage.java

```diff
@@ -71,30 +55,30 @@ final class DetailMessage extends TestDiagnostic {
   final String readableMessage;
 
   /**
-   * Returns an object parsed from a diagnostic message, or {@code null} if the message doesn't
-   * match the expected format.
+   * Returns an object parsed from a diagnostic message.
    *
    * @param rootDirectory if not null, a root directory prefix to remove from the file part of the
    *     message
    */
-  static @Nullable DetailMessage parse(String input, @Nullable Path rootDirectory) {
-    Matcher messageMatcher = MESSAGE_PATTERN.matcher(input);
-    if (!messageMatcher.matches()) {
-      return null;
-    }
-
-    Path file = Paths.get(messageMatcher.group("file"));
-    if (rootDirectory != null) {
+  static DetailMessage parse(TestDiagnostic input, @Nullable Path rootDirectory) {
+    Path file = input.getFile();
+    if (rootDirectory != null && !file.toString().equals("")) {
```

Comment:
```suggestion
    if (rootDirectory != null && file.startsWith(rootDirectory)) {
```

I think this is more robust. I think `relativize()` throws if `file` isn't within `rootDirectory`.